zerver/actions/presence.py

```diff
@@ -82,7 +83,7 @@ def consolidate_client(client: Client) -> Client:
 def do_update_user_presence(
     user_profile: UserProfile,
     client: Client,
-    log_time: datetime.datetime,
+    log_time: Optional[datetime.datetime],
```

Comment:
I don't understand the motivation for making `log_time` optional instead of just having the test pass a value for this -- can you explain your thinking?

Reply:
I was thinking that this method is accepting `log_time` as `None`. But it is not being called with `None` value anywhere inside the code. So, should we remove the `Optional` type for it?

Reply:
I have updated the PR now @timabbott